### PR TITLE
schedulers: add blacklist with store limit

### PR DESCRIPTION
**Pull request by lhy1024**
Signed-off-by: lhy1024 <admin@liudos.us>

### What problem does this PR solve? 

it will try to solve #2071 and #2002 at the same time. This PR will restore black list after #1823.

### What is changed and how it works?

- [x] merge #2066
- [ ] test on default limit and unlimit

### Check List 

Tests 

 - Unit test

### CODE REVIEW

File: server/schedulers/balance_region.go [hunk 1]
```diff
@@ -130,6 +144,7 @@ func (s *balanceRegionScheduler) IsScheduleAllowed(cluster opt.Cluster) bool {
 func (s *balanceRegionScheduler) Schedule(cluster opt.Cluster) []*operator.Operator {
 	schedulerCounter.WithLabelValues(s.GetName(), "schedule").Inc()
 	stores := cluster.GetStores()
+	s.filters[1] = s.hitsCounter.buildSourceFilter(s.GetName(), cluster)
```
Comment (disksing) [inline server/schedulers/balance_region.go]: kind strange to reset filters[1] every time. I prefer `filters := append(s.filters, s.hitsCounter.buildSourceFilter(s.GetName(), cluster))`

Comment (lhy1024) [reply server/schedulers/balance_region.go]: done

File: server/schedulers/balance_region.go [hunk 2]
```diff
@@ -241,3 +261,130 @@ func (s *balanceRegionScheduler) transferPeer(cluster opt.Cluster, region *core.
...
+		if sourceTTL, ok := h.ttl[store.GetID()]; ok && sourceTTL < ttl {
+			ttl = sourceTTL
+		}
+	}
+	return ttl
+}
+
+func (h *hitsStoreBuilder) updateTTL(limits map[uint64]float64) {
+	for storeID, limit := range limits {
+		h.ttl[storeID] = time.Second * time.Duration(limit*hitLimitFactor)
```
Comment (rleungx) [inline server/schedulers/balance_region.go]: How about add some comments to explain why `time.Second * time.Duration(limit*hitLimitFactor)`?

File: server/schedulers/balance_test.go [hunk 3]
```diff
@@ -1308,3 +1309,31 @@ func (s *testScatterRangeLeaderSuite) TestBalanceWhenRegionNotHeartbeat(c *C) {
...
+}
+
+func (s *testTTL) getStoreLimits(stores uint64, limit float64) map[uint64]float64 {
+	limits := make(map[uint64]float64)
+	for i := uint64(1); i < stores; i++ {
+		limits[i] = limit
+	}
+	return limits
+}
+func (s *testTTL) checkTTL(c *C, ttl map[uint64]time.Duration, limit float64) {
```
Comment (rleungx) [inline server/schedulers/balance_test.go]: add a blank line?

**Pull request closed without merging**